src/platform/windows/display_uwp.cpp

```diff
@@ -0,0 +1,247 @@
+/**
+ * @file src/platform/windows/display_uwp.cpp
+ * @brief WinRT Windows.Graphics.Capture API
+ */
+#include <dxgi1_2.h>
+
+#include "display.h"
+
+#include "misc.h"
+#include "src/logging.h"
+
+#include <windows.graphics.capture.interop.h>
+#include <winrt/windows.foundation.h>
+#include <winrt/windows.graphics.directx.direct3d11.h>
+
+namespace platf {
+  using namespace std::literals;
+}
+
+namespace winrt {
+  using namespace Windows::Foundation;
+  using namespace Windows::Graphics::Capture;
+  using namespace Windows::Graphics::DirectX::Direct3D11;
+
+  extern "C" {
+  HRESULT __stdcall CreateDirect3D11DeviceFromDXGIDevice(::IDXGIDevice *dxgiDevice, ::IInspectable **graphicsDevice);
+  }
+  struct
+#if WINRT_IMPL_HAS_DECLSPEC_UUID
+    __declspec(uuid("A9B3D012-3DF2-4EE3-B8D1-8695F457D3C1"))
+#endif
+      IDirect3DDxgiInterfaceAccess: ::IUnknown {
+    virtual HRESULT __stdcall GetInterface(REFIID id, void **object) = 0;
+  };
+}  // namespace winrt
+#if !WINRT_IMPL_HAS_DECLSPEC_UUID
+static constexpr GUID GUID__IDirect3DDxgiInterfaceAccess = {
+  0xA9B3D012, 0x3DF2, 0x4EE3, { 0xB8, 0xD1, 0x86, 0x95, 0xF4, 0x57, 0xD3, 0xC1 }
+  // compare with __declspec(uuid(...)) for the struct above.
+};
+template <>
+constexpr auto
+__mingw_uuidof<winrt::IDirect3DDxgiInterfaceAccess>() -> GUID const & {
+  return GUID__IDirect3DDxgiInterfaceAccess;
+}
+#endif
+
+namespace platf::dxgi {
+  static HRESULT
+  create_item_for_monitor(HMONITOR mon, winrt::GraphicsCaptureItem &item) {
+    auto interop_factory = winrt::get_activation_factory<winrt::GraphicsCaptureItem, IGraphicsCaptureItemInterop>();
+    return interop_factory->CreateForMonitor(mon, winrt::guid_of<winrt::Windows::Graphics::Capture::IGraphicsCaptureItem>(), winrt::put_abi(item));

```

Comment:
This is only present for Win11 and Server 2022. What happens on Windows 10 if this is invoked?

(same with a few othe WGC APIs like `IsBorderRequired`)

Reply:
💀
(Note to self: `GraphicsCaptureItem::TryCreateFromDisplay` has the same level of support as `IsBorderRequired`, use instead)

If display creation fails, or (I suppose if this interop factory, which I'll be removing, is null), then I plan to have the display backend initializer return -1 so control falls to the original API.

Reply:
for future posterity, `GraphicsCaptureItem::TryCreateFromDisplayId` is a nogo, probably:

https://learn.microsoft.com/en-us/uwp/api/windows.graphics.capture.graphicscaptureitem.trycreatefromdisplayid?view=winrt-22621
>Before attempting to create a GraphicsCaptureItem with this method, you must call [GraphicsCaptureAccess.RequestAccessAsync](https://learn.microsoft.com/en-us/uwp/api/windows.graphics.capture.graphicscaptureaccess.requestaccessasync?view=winrt-22621#windows-graphics-capture-graphicscaptureaccess-requestaccessasync(windows-graphics-capture-graphicscaptureaccesskind)), passing in a value of [GraphicsCaptureAccessKind.Programmatic](https://learn.microsoft.com/en-us/uwp/api/windows.graphics.capture.graphicscaptureaccesskind?view=winrt-22621). To call RequestAccessAsync with GraphicsCaptureAccessKind.Programmatic, you must declare the graphicsCaptureProgrammatic capability in your app's package manifest. For more information, see [App capability declarations](https://learn.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations).

The way I am currently doing this is _probably_ fine, since we can fall back to DXGI DDA if this isn't present.